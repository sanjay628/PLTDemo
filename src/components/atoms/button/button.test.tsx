import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import Button from './button.component';
const createTestProps = (props: Object) => ({
  ...props,
});
describe('Button', () => {
  let props: any;
  beforeEach(() => {
    props = createTestProps({});
  });
  it('renders button with label and calls onPress handler', () => {
    const onPressMock = jest.fn();
    const {getByText} = render(
      <Button
        btnLabel="Submit"
        onPress={onPressMock}
        buttonTextColor={'red'}
        buttonColor={'red'}
      />,
    );
    const buttonElement = getByText('Submit');
    expect(buttonElement).toBeDefined();
    fireEvent.press(buttonElement);
    expect(onPressMock).toHaveBeenCalled();
  });

  it('renders disabled button', () => {
    const onPressMock = jest.fn();
    const {getByText} = render(
      <Button
        btnLabel="Submit"
        onPress={onPressMock}
        disable={true}
        buttonTextColor={'red'}
        buttonColor={'red'}
      />,
    );
    const buttonElement = getByText('Submit');
    expect(buttonElement).toBeDefined();
    fireEvent.press(buttonElement);
    expect(onPressMock).not.toHaveBeenCalled();
  });

  it('applies custom styles and testID', () => {
    const onPressMock = jest.fn();
    const {getByTestId} = render(
      <Button
        btnLabel="Submit"
        onPress={onPressMock}
        buttonStyle={{borderRadius: 8}}
        buttonTextColor={'red'}
        buttonColor={'red'}
      />,
    );
    const buttonElement = getByTestId('submit-button');
    expect(buttonElement).toBeDefined();
  });

  it('renders correctly', () => {
    const tree = render(<Button {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
