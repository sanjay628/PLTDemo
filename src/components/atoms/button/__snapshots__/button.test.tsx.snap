// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": undefined,
      "borderRadius": 40,
      "elevation": 5,
      "height": 82,
      "justifyContent": "center",
      "opacity": 1,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 8,
        "width": 8,
      },
      "shadowOpacity": 0.6,
      "shadowRadius": 10,
      "width": undefined,
    }
  }
  testID="submit-button"
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 32,
          "fontWeight": "600",
        },
        undefined,
        Object {
          "color": undefined,
        },
      ]
    }
  />
</View>
`;
